src/m365/planner/commands/plan/plan-set.ts

```diff
@@ -0,0 +1,156 @@
+    return 'Set a Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (error?: any) => void): void {
+    const getPlan = (): Promise<CommandOutput> => {
+      if (this.verbose) {
+        logger.logToStderr(`Retrieving plan ${args.options.id || args.options.title} ...`);
+      }
+
+      const options: PlanGetCommandOptions = {
+        ...args.options,
+        output: 'json',
+        debug: this.debug,
+        verbose: this.verbose
+      };
+
+      return Cli.executeCommandWithOutput(planGetCommand as Command, { options: { ...options, _: [] } });
+    };
+
+    const updatePlan = (plan: any): Promise<void> => {
+      if (this.verbose) {
+        logger.logToStderr(`Updating plan with id ${plan['id']} ...`);
+      }
+
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/planner/plans/${plan['id']}`,
+        headers: {
+          'accept': 'application/json',
+          'If-Match': `${plan["@odata.etag"]}`
+        },
+        responseType: 'json',
+        data: {
+          title: args.options.newTitle
+        }
+      };
+
+      return request.patch(requestOptions);
+    };
+
+    getPlan()
+      .then((output: CommandOutput): void => {
+        if (!output.stdout) {
+          cb(new CommandError(`No plan found`));
+          return;
+        }
+        let plans: any = JSON.parse(output.stdout);
+        if(!Array.isArray(plans)) {
+          plans = [plans];
+        }
+        const promises = [];
+        for (let index = 0; index < plans.length; index++) {
+          promises.push(updatePlan(plans[index]));
+        }
+
+        Promise.all(promises).then(() => {
+          cb();
+        })
+          .catch((err) => {
+            this.handleRejectedODataJsonPromise(err, logger, cb);
+          });
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-t, --title [title]'
+      },
+      {
+        option: '--newTitle <newTitle>'
+      },
+      {
+        option: '--ownerGroupId [ownerGroupId]'
+      },
+      {
+        option: '--ownerGroupName [ownerGroupName]'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (!args.options.newTitle) {
```

Comment:
We check that required options have been set centrally in the CLI so we don't need to do it here